analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/EmptyCollectionsShouldNotBeEnumeratedBase.cs

```diff
@@ -161,4 +148,33 @@ public override void ExecutionCompleted()
 
     private static bool IsEmptyArray(IOperation operation) =>
         operation.AsArrayCreation()?.DimensionSizes.Any(x => x.ConstantValue.Value is 0) ?? false;
+
+    private ProgramState ProcessMethod(SymbolicContext context, IMethodSymbol method, IOperation instance)
+    {
+        if (instance is null)
+        {
+            return context.State;
+        }
+        if (RaisingMethods.Contains(method.Name))
+        {
+            if (context.State[instance]?.HasConstraint(CollectionConstraint.Empty) is true)
+            {
+                emptyAccess.Add(context.Operation.Instance);
+            }
+            else
+            {
+                nonEmptyAccess.Add(context.Operation.Instance);
+            }
+        }
+        var state = context.State;
```

Comment:
I would move this at the beginning because using `state` and `context.State` in the same method in different places can lead to incidents later.